### PR TITLE
chore: Upgrade spectral dependency

### PR DESCRIPTION
**Pull request by filfreire**
## Changes

- Upgraded `@spotlight/spectral` dependency to latest versions. There's been some major changes on Spectral, and the latest version[ is no longer published in the same package](https://www.npmjs.com/package/@stoplight/spectral). Now we need to use specific packages directly, like `spectral-core`, `spectral-formats` and `spectral-rulesets`.
- Added a smoke check where we import our default Petstore OpenAPI example and we try to force an OpenAPI lint error.

## How to test locally

- Checkout this branch
- Run `npm run bootstrap`
- Run `npm run watch:app`
- On a 2nd terminal window run `npm run test:smoke:dev -- openapi`

### CODE REVIEW

File: packages/insomnia-inso/src/commands/lint-specification.ts [hunk 1]
```diff
@@ -50,9 +51,7 @@ export async function lintSpecification(
   }
 
   const spectral = new Spectral();
-  spectral.registerFormat('oas2', isOpenApiv2);
-  spectral.registerFormat('oas3', isOpenApiv3);
-  await spectral.loadRuleset('spectral:oas');
+  await spectral.setRuleset(oas as unknown as Ruleset);
```
Comment (jackkav) [inline packages/insomnia-inso/src/commands/lint-specification.ts]: could be a ts-ignore?

Comment (filfreire) [reply packages/insomnia-inso/src/commands/lint-specification.ts]: Yeah, seems to work 👍

**Review by jackkav — APPROVED** [review]
🚀

**Comment by DMarby** [comment]
Does this mean that we can get rid of our `immer` dependency workaround as well? (ping @dimitropoulos) https://github.com/Kong/insomnia/blob/cc9f11be30f365fdc70cd96bab5fa48a591f2415/package.json#L68-L70

**Comment by dimitropoulos** [comment]
yes, absolutely we can get rid of the immer hack now and (imo) should do so in this PR since the spectral upgrade was the only blocker.

File: packages/insomnia-inso/src/commands/lint-specification.ts [hunk 2]
```diff
@@ -50,9 +51,8 @@ export async function lintSpecification(
   }
 
   const spectral = new Spectral();
-  spectral.registerFormat('oas2', isOpenApiv2);
-  spectral.registerFormat('oas3', isOpenApiv3);
-  await spectral.loadRuleset('spectral:oas');
+  // @ts-expect-error -- TSCONVERSION oas not being properly caught as Ruleset
```
Comment (dimitropoulos) [inline packages/insomnia-inso/src/commands/lint-specification.ts]: please do not mark things as TSCONVERSION, that was a sortof one-time thing for the time when we converted to typescript.  all errors from here forward should be handled.  I'll try to take a look at this one.  I remember hitting it before when I upgraded spectral last year.

Comment (filfreire) [reply packages/insomnia-inso/src/commands/lint-specification.ts]: Understood. I went ahead and removed the `TSCONVERSION` keyword for this `@ts-expect-error` message.

**Comment by filfreire** [comment]
> Does this mean that we can get rid of our `immer` dependency workaround as well? (ping @dimitropoulos)
> 
> https://github.com/Kong/insomnia/blob/cc9f11be30f365fdc70cd96bab5fa48a591f2415/package.json#L68-L70

@DMarby @dimitropoulos I went ahead and also removed this `immer` dependency 👍

File: packages/insomnia-inso/src/commands/lint-specification.ts (hunk 2 above)
Comment (dimitropoulos) [reply packages/insomnia-inso/src/commands/lint-specification.ts]: you removed the other one in spectral.ts it looks like (which also needed to be removed).

Comment (dimitropoulos) [reply packages/insomnia-inso/src/commands/lint-specification.ts]: fixed in 22fa1a3114c01dab4f3b2401d617abbc9ed68d4c

**Pull request closed without merging**